src/features/boardSlice.ts

```diff
@@ -0,0 +1,28 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { v4 as uuidv4 } from "uuid";
+import Board from "../types/global/Board";
+import BoardState from "../types/reducers/BoardSlice";
+
+const boardData: Board = JSON.parse(localStorage.getItem("board") || "{}");
```

Comment:
When you load something using localStorage.getItem, you may get a null. JSON.parse expects non null data, hence the or with a string of empty object